extensions/MozChangeField/lib/Post/RegressedByTypeKeyword.pm

```diff
@@ -0,0 +1,77 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+# This Source Code Form is "Incompatible With Secondary Licenses", as
+# defined by the Mozilla Public License, v. 2.0.
+
+package Bugzilla::Extension::MozChangeField::Post::RegressedByTypeKeyword;
+
+use 5.10.1;
+use Moo;
+
+use Bugzilla::Keyword;
+
+sub evaluate_create {
+  my ($self, $args) = @_;
+  my $bug       = $args->{bug};
+  my $timestamp = $args->{timestamp};
+  my $dbh       = Bugzilla->dbh;
+
+  if (@{$bug->regressed_by}) {
+    if (!$bug->has_keyword('regression')) {
+      my $keyword_obj = Bugzilla::Keyword->new({name => 'regression'});
+      if ($keyword_obj) {
+        $dbh->do('INSERT INTO keywords (bug_id, keywordid) VALUES (?, ?)',
+          undef, $bug->id, $keyword_obj->id);
+        delete $bug->{'keyword_objects'};    # Clear current keywords so they have to be reloaded.
+      }
+    }
+
+    if ($bug->bug_type ne 'defect') {
+      $dbh->do('UPDATE bugs SET bug_type = ? WHERE bug_id = ?',
+        undef, 'defect', $bug->id);
+      $bug->{bug_type} = 'defect';
+    }
+  }
+}
+
+sub evaluate_change {
+  my ($self, $args) = @_;
+  my $bug       = $args->{bug};
+  my $timestamp = $args->{timestamp};
+  my $changes   = $args->{changes};
+  my $dbh       = Bugzilla->dbh;
+
+  if (@{$bug->regressed_by} && exists $changes->{regressed_by}) {
```

Comment:
Nit; all the code for this function is in the main `if` statement. Instead of having `if (condition)` with all the code inside the block, we should do `if (not condition) {return}` and then put everything else below, de-dented.